lib/LaTeXML/Package/expl3.lua.ltxml

```diff
@@ -0,0 +1,124 @@
+# |  United States Government & not subject to copyright in the US.     | #
+# |---------------------------------------------------------------------| #
+# | Bruce Miller <bruce.miller@nist.gov>                        #_#     | #
+# | http://dlmf.nist.gov/LaTeXML/                              (o o)    | #
+# \=========================================================ooo==U==ooo=/ #
+package LaTeXML::Package::Pool;
+use strict;
+use warnings;
+use LaTeXML::Package;
+
+# Translation of the intarray portion of expl3.lua
+# lua arrays are typically 1-based
+# Apparently an expl3 intarray is encoded at the TeX level by \__intarray:w <Number>
+# We store actual ints in the array, seems more efficient (could be Number or Tokens)
+DefParameterType('Intarray', sub {
+    my ($gullet) = @_;
+    my $tok = $gullet->readXToken;
+    if (my $defn = LookupDefinition($tok)) {    # token may be \let to the identifier!!!
+      $tok = $defn->getCS; }
+    if ($tok && $tok->equals(T_CS('\__intarray:w'))) {
+      my $n       = $gullet->readNumber;
+      my $address = '__intarray' . '_' . ToString($tok) . '_' . $n->valueOf;
+      my $table   = LookupValue($address);
+      if (!$table) {                            # Create new table, if none
+        $table = [];
+        AssignValue($address => $table, 'global'); }
+      return $table; }
+    else {
+      $gullet->unread($tok);
+      Error('expected', '__intarray:w', $gullet,
+        "Expected an intarray identifier, got " . ToString($tok));
+      return []; } });
+
+DefPrimitiveI('\__intarray:w', 'Number', sub {
+    Error('unexpected', '\__intarray:w', $_[0], "Unexpected isolated \\__intarray:w?");
+    return; },
+  protected => 1);
+
+DefPrimitiveI('\__intarray_gset_count:Nw', 'Intarray Number', sub {
+    my ($stomach, $table, $newlength) = @_;
+    my $length = scalar(@$table);
+    $newlength = $newlength->valueOf;
+    $$table[$newlength - 1] = 0;                     # Grow in one step for efficiency
+    for (my $i = $length + 1 ; $i <= $newlength ; $i++) {
+      $$table[$i - 1] = 0; }
+    return; },
+  protected => 1);
+
+DefMacroI('\intarray_count:N', 'Intarray', sub {
+    my ($gullet, $table) = @_;
+    return Explode(scalar(@$table)); });
+
+DefMacroI('\__intarray_gset:wF', 'Number Intarray Number {}', sub {
+    my ($stomach, $pos, $table, $value, $ifmissing) = @_;
+    my $length = scalar(@$table);
+    $pos = $pos->valueOf;
+    if (($pos > 0) && ($pos < $length)) {
+      $$table[$pos - 1] = $value->valueOf; }
+    else {
+      return $ifmissing; }
+    return; },
+  protected => 1);
+
+DefMacroI('\__intarray_gset:w', 'Number Intarray Number', sub {
+    my ($stomach, $pos, $table, $value) = @_;
+    my $length = scalar(@$table);
+    $pos = $pos->valueOf;
+    if ($pos > 0) {
+      $$table[$pos - 1] = $value->valueOf; }
+    return; },
+  protected => 1);
+
+DefPrimitiveI('\intarray_gzero:w', 'Intarray', sub {
```

Comment:
My dtx file is defining `\intarray_gzero:N` at this point, and no `\intarray_gzero:w`.

Reply:
I wasn't completely clear on the lua interface, so I just implemented (or didn't) everything related to `intarray`.